tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -274,48 +274,45 @@ def access_string_vhdl(dut):
 # TODO: add tests for Verilog "string_input_port" and "STRING_LOCALPARAM" (see issue #802)
 
 @cocotb.test(skip=cocotb.LANGUAGE in ["vhdl"] or cocotb.SIM_NAME.lower().startswith("icarus"),
-             expect_fail=cocotb.SIM_NAME.lower().startswith("modelsim"),
-             expect_error=cocotb.SIM_NAME.startswith(("xmsim", "ncsim")))
+             expect_error=cocotb.result.TestFailure if cocotb.SIM_NAME.lower().startswith(("xmsim", "ncsim", "modelsim", "chronologic simulation vcs")) else False)
 def access_const_string_verilog(dut):
     """Access to a const Verilog string."""
     tlog = logging.getLogger("cocotb.test")
-
-    yield Timer(10)
     string_const = dut.STRING_CONST;
+
+    yield Timer(10, 'ns')
     tlog.info("%r is %s" % (string_const, str(string_const)))
     if not isinstance(string_const, StringObject):
         raise TestFailure("STRING_CONST was not StringObject")
-    if string_const != "TESTING_CONST":
-        raise TestFailure("STRING_CONST was not == \'TESTING_CONST\'")
+    if string_const != b"TESTING_CONST":
+        raise TestFailure("Initial value of STRING_CONST was not == b\'TESTING_CONST\' but {} instead".format(string_const))
 
     tlog.info("Modifying const string")
-    string_const <= "MODIFIED"
-    yield Timer(10)
-    string_const = dut.STRING_CONST;
-    if string_const != "TESTING_CONST":
-        raise TestFailure("STRING_CONST was not still \'TESTING_CONST\'")
+    string_const <= b"MODIFIED"
+    yield Timer(10, 'ns')
+    if string_const != b"TESTING_CONST":
+        raise TestFailure("STRING_CONST was not still b\'TESTING_CONST\' after modification but {} instead".format(string_const))
```

Comment:
This duplicates #1381 I assume, which should be merged first.

Reply:
Yes, I'll keep this as a draft until it is in.